gems/pending/appliance_console/date_time_configuration.rb

```diff
@@ -0,0 +1,96 @@
+require "appliance_console/logging"
+require 'appliance_console/prompts'
+
+module ApplianceConsole
+  class DateTimeConfiguration
+    DATE_REGEXP   = /^(2[0-9]{3})-(0?[1-9]|1[0-2])-(0?[1-9]|[12][0-9]|3[01])/
+    DATE_PROMPT   = "current date (YYYY-MM-DD)".freeze
+    TIME_REGEXP   = /^(0?[0-9]|1[0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9])/
+    TIME_PROMPT   = "current time in 24 hour format (HH:MM:SS)".freeze
+
+    attr_accessor :new_date, :new_time
+
+    include ApplianceConsole::Logging
+
+    def initialize(config = {})
+      @new_date = config[:new_date]
+      @new_time = config[:new_time]
+    end
+
+    def activate
+      say("Applying time configuration...")
+      disable_auto_sync &&
+        configure_date_time
+    end
+
+    def ask_questions
+      ask_disable_auto_sync &&
+        ask_for_date &&
+        ask_for_time &&
+        confirm
+    end
+
+    def ask_disable_auto_sync
+      say("Automatic time synchronization must be disabled to manually set date or time\n\n")
+
+      return false unless agree("Disable Automatic time synchronization? (Y/N): ")
+      say("Automatic time synchronization will be disabled...")
+      true
+    end
+
+    def ask_for_date
+      begin
+        @new_date = just_ask(DATE_PROMPT, nil, DATE_REGEXP)
+      rescue
+        return false
+      end
+      true
+    end
+
+    def ask_for_time
+      begin
+        @new_time = just_ask(TIME_PROMPT, nil, TIME_REGEXP)
+      rescue
+        return false
+      end
+      true
+    end
+
+    def confirm
+      clear_screen
+      say(<<-EOL)
+Date and Time Configuration
+
+      Date:          #{new_date}
+      Time:          #{new_time}
+
+        EOL
+
+      return false unless agree("Apply time configuration? (Y/N): ")
+      true
+    end
+
+    def disable_auto_sync
+      begin
+        LinuxAdmin::Service.new("chronyd").stop.disable
+        LinuxAdmin::Service.new("systemd-timedated").restart
+      rescue => e
+        say("Failed to disable time synchronization")
+        Logging.logger.error("Failed to disable time synchronization: #{e.message}")
+        return false
+      end
+      true
+    end
+
+    def configure_date_time
+      begin
+        LinuxAdmin::TimeDate.system_time = Time.parse("#{new_date} #{new_time}").getlocal
```

Comment:
http://ruby-doc.org/stdlib-2.2.4/libdoc/time/rdoc/Time.html#method-c-parse seems to say that we should get something in the current timezone if we don't specify a second parameter. So I think this will be fine either way.
>If the upper components of the given time are broken or missing, they are supplied with those of `now`.